accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,338 @@
+  - From that list of feature bands, select the most recent one prior to the current SDK Feature band
+  - If such a feature band exists, then get the list of workloads installed for that feature band, and use that as the list of workloads to install.  Otherwise, there is nothing to update, and quit the update process
+- With rollback:
+  - Install updated workload manifests
+  - Run workload installation process for list of workloads that were installed
+- If unit of installation is workload pack:
+  - Run garbage collection to remove leftover workload packs
+- If unit of installation is workload:
+  - The IAL should handle uninstalling previous workloads from the same SDK feature band (or treating the install of a newer version as a patch of the previous ones)
+
+# Installer Abstraction API
+
+- Get *unit of installation*: workloads or packs
+- If unit of installation is workloads
+  - Install workload
+    - Optionally using offline installation cache folder
+    - Should support transaction / rollback
+  - Download workload artifacts to offline cache folder (for a given workload)
+    - This should support using the currently installed workload manifests, or another set of manifest, which may be for a future SDK feature band.  This is so that VS for Mac can use the currently installed version of the SDK to pre-download the assets for an updated SDK.
+  - Uninstall workload
+  - List installed workloads
+  - Not part of the API, but important for implementation: How to map from SDK workload ID to underlying installation ID and version?
+- If unit of installation is workload pack
+  - Install workload pack
+    - Optionally using offline installation cache folder
+    - Should support transaction / rollback
+  - Download workload pack artifacts to offline cache folder
+    - This should support using the currently installed workload manifests, or another set of manifest, which may be for a future SDK feature band.
+  - Garbage collect installed workload packs
+- Install workload manifest (given a workload manifest ID and version)
+  - Should support transaction / rollback
+  - This may involve downloading a package or installer for the workload manifest
+  - Should optionally support using offline installation cache folder
+  - End result should be that the manifest under `<DOTNET ROOT>/sdk-manifests/<SdkFeatureBand>/` is installed / updated
+- Does IAL support installing or just giving instructions on how to install?
+  - Initial installer implementations may all support installation directly, so this may not need to be included in the first version of the installer abstraction API
+- Workload installation records
+  - List installed workloads (SDK feature band)
+  - Write installation record (workload ID, SDK feature band)
+  - Delete installation record (workload ID, SDK feature band)
+  - List feature bands that have workload installation records
+
+# Installer implementations
```

Comment:
Our customers build their container apps in containers using an SDK image.
The SDK image we build, can only contain things we can build from source.

For workloads that would be typical for microservices, we want to be able to source-build and include them with our SDK. Otherwise each application build would need to start by installing a workload pack.

Reply:
For .NET 6, the workloads will be for iOS, Android, .NET MAUI, and Blazor AOT support.  I think that for .NET 6, iOS, Android, and .NET MAUI won't be supported at all on Linux.  So that only leaves Blazor AOT that would be relevant to you for .NET 6.

Reply:
Source build for Blazor AOT will be likely be a long term item.

Reply:
@dsplaisted yes, Blazor AOT would be the one we care most about.
@lewing can you create an issue in the appropriate repository to track support for source-building Blazor AOT? I have no idea what this involves doing. Can you include some information on the ticket?